lib/msal-browser/src/config/Configuration.ts

```diff
@@ -278,6 +282,7 @@ export function buildConfiguration(
         },
         skipAuthorityMetadataCache: false,
         supportsNestedAppAuth: false,
+        extraTokenRequestHeaders: {},
```

Comment:
Would this be more appropriate under `system`? I also wonder if this is the kind of functionality that overriding the network client would be better suited for?

Reply:
I don't have a preference. I put it under auth because it's part of the request logic, I'm not particularly married to that. As far as falling back on the network client override, the only downside I can think of is changing the headers per-request wouldn't be as easy.

Reply:
There's also more overhead if overriding the network client is required.

Reply:
What's the scenario we're trying to unblock today? Does it require request level headers?

Reply:
No immediate scenario, mostly we're adding this so we don't have to ship a new version later if a new header is required for some exception (things like browser feature trials, for ex. Chrome's 3p cookie extension trial).

Reply:
This is more a just in case we need to add something in here.  No active scenario we need to unblock at the moment.

Reply:
TLDR: Let's punt this unless we need/have a requirement. My POV below:

Historically we rarely had any new STS params added (except may be the format change for `login_hint`.) Do we foresee this happening for headers? I would like to get STS involved once to check how critical is this and/or other MSALs if they support this and also check if it is applicable for browsers. Adding in configuration especially, I am very mindful of exposing this to customers for params we do not support. If the STS ignores this, it is all fine but if the service throws an error, we are opening ourselves for something unnecessary. This in turn would then open this up for sanitization and the lack of use case does not render us enough material to do so.